### ISSUE TITLE
closing connection manually does not trigger onClose

### ISSUE THREAD

**Issue by Stevemoretz** [ratchetphp/Ratchet#948]
### What I've figured out so far

1. The connection is reported to be closed for client with the same 1010 code, so it actually gets closed and no problem on that side.
2. If the connection is closed by the client, this function gets triggered without any problems.

### The problem
I'm doing something like:

```php
$conn->close(1010);
```

But this function is not triggered.
```php
    public function onClose( ConnectionInterface $conn ) {
        $this->print("ssss");
        parent::onClose( $conn );
    }
```

I use that function to cleanUp connection specific data, so it's kind of important.
however I can just call it myself, but I need to know is this a bug or is it intended? (to prevent duplication.)

**Comment by inder172** [comment]
I am Facing Same Problem.did you Find any Solution?
@Stevemoretz

**Comment by Stevemoretz** [comment]
> I am Facing Same Problem.did you Find any Solution? @Stevemoretz

The workaround is to extend the class, override the close method and call onClose in there.

**Comment by inder172** [comment]
Can you Please share the Code.

actually I am using enablekeepAlive and it is not Suitable for Large number of websocket Connection

**Comment by Stevemoretz** [comment]
You're right my explaining was insufficient, actually it has one more step, you need to create a client wrapper, also you need to put your ratchet server instance in a container or global variable, once you call the close on your client wrapper you need to access the global variable and call onClose in there manually.

I can't share my code because it has a lot of dependency subclasses it would be harder to Understand, but that's the way it works under the hood, if you have any further questions feel free to ask.

**Comment by PaulRotmann** [comment]
Hey! Thanks for reporting this issue 🙏

We're giving Ratchet some much-needed love and attention! As part of our issue cleanup initiative ([#1100](https://github.com/ratchetphp/Ratchet/issues/1100)), we're reviewing and closing issues that look like they might already be resolved.

If this is still relevant to you, we'd love to continue this conversation in our [GitHub Discussions](https://github.com/ratchetphp/Ratchet/discussions). Just reference this issue when you post. If you want to support Ratchet, please consider [sponsoring our work](https://github.com/sponsors/reactphp)! ❤️
